### PR TITLE
Complete steps using Main's isRandomEnabled

### LINKED ISSUE #225: Create `disable_random` command handler

### PR DESCRIPTION
**Pull request by triszt4n**
Closes #225 

disable_random command was already done, but the ISteppable-s didn't use them yet. Now it's okay.

### CODE REVIEW

File: java/src/hu/holyoil/controller/AIController.java [hunk 1]
```diff
@@ -35,7 +36,13 @@
      */
     @Override
     public void Step()  {
-        System.out.println("Stepping");
+        Logger.Log(this, "Steps" + (Main.isRandomEnabled ? "" : " - switched off!"));
+
+        if (Main.isRandomEnabled) {
```
Comment (vercibar) [inline java/src/hu/holyoil/controller/AIController.java:41]: Randomness disabling should be in the classes themselves

File: java/src/hu/holyoil/controller/GameController.java [hunk 2]
```diff
@@ -32,7 +33,14 @@
      */
     @Override
     public void Step() {
-        System.out.println("Stepping");
+        Logger.Log(this, "Steps" + (Main.isRandomEnabled ? "" : " - switched off!"));
+
+        if (Main.isRandomEnabled) {
```
Comment (vercibar) [inline java/src/hu/holyoil/controller/GameController.java:38]: Randomness does not effect win and lose conditions.

File: java/src/hu/holyoil/controller/SunController.java [hunk 3]
```diff
@@ -38,18 +39,34 @@ public String toString() {
...
+    private int turnsUntilNextSunstorm;
     /**
      * A pályán található összes aszteroidát tartalmazó lista
      */
     private List<Asteroid> asteroids;
 
+    /**
+     * Újraindítja a visszaszámláló turnsUntilNextSunstorm számlálót egy random számra 20 és 50 között
+     */
+    private void RestartCountdown() {
```
Comment (vercibar) [inline java/src/hu/holyoil/controller/SunController.java:51]: This should account for randomness enabled

File: java/src/hu/holyoil/controller/SunController.java [hunk 4]
```diff
@@ -38,18 +39,34 @@ public String toString() {
...
+
     /**
      * minden körben végrehajt egy lépést
      */
     @Override
     public void Step() {
-        System.out.println("Stepping");
+        Logger.Log(this, "Steps" + (Main.isRandomEnabled ? "" : " - switched off!"));
+
+        if (Main.isRandomEnabled) {
```
Comment (vercibar) [inline java/src/hu/holyoil/controller/SunController.java:63]: This does not change when randomness is disabled

**Review by vercibar — CHANGES_REQUESTED** [review]
I believe randomness should be handled differently. Currently disabling randomness makes the game not function.

File: java/src/hu/holyoil/controller/GameController.java (hunk 2 above)
Comment (triszt4n) [reply java/src/hu/holyoil/controller/GameController.java:38]: lol true

File: java/src/hu/holyoil/controller/SunController.java (hunk 3 above)
Comment (triszt4n) [reply java/src/hu/holyoil/controller/SunController.java:52]: Done

File: java/src/hu/holyoil/controller/AIController.java (hunk 1 above)
Comment (triszt4n) [reply java/src/hu/holyoil/controller/AIController.java]: Well, we discussed, we don't want the AI to handle Ufo and Robot at all, if the disable_random is called.
So I moved the conditional to the Handle methods. But right now the teleportgates have this conditional too. What do I do with that? Stays liek that?

File: java/src/hu/holyoil/controller/SunController.java (hunk 4 above)
Comment (triszt4n) [reply java/src/hu/holyoil/controller/SunController.java]: Done

**Review by vercibar — APPROVED** [review]

**Pull request merged**